app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,80 @@
+class TasksController < ApplicationController
+  
+  load_and_authorize_resource
+  def index
+      @tasks = @tasks.paginate(page: params[:page], per_page: 2)
+  end
+
+  def show
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def edit
+  end
+
+  def create
+    @task = Task.new(task_params)
+    respond_to do |format|
+      if @task.save
+        format.html { redirect_to @task, notice: 'Task was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @task.update(task_params)
+        format.html { redirect_to @task, notice: 'Task was successfully updated.' }
+      else
+        format.html { render :edit }
+      end
+    end
+  end
+
+  def destroy
+    respond_to do |format|
+      if @task.destroy
+        format.html { redirect_to tasks_url, notice: 'Task was successfully destroyed.' }
+      else
+        format.html { redirect_to tasks_url, notice: 'Task cannot destroy!' }
```

Comment:
send error
redirect to tasks_path (diff between path and url)
in case of failure redirect to task show